app/apps3buckets/handlers.js

```diff
@@ -1,54 +1,55 @@
 const { ensureLoggedIn } = require('connect-ensure-login');
-
-const api = require('../api-client');
-const routes = require('../routes');
+const { App, AppS3Bucket } = require('../api-client');
 
 
 exports.create = [
   ensureLoggedIn('/login'),
   (req, res, next) => {
     const { app_id, bucket_id } = req.body;
 
-    const apps3bucket = {
-      app: app_id,
-      s3bucket: bucket_id,
-      access_level: 'readonly',
-    };
+    App.get(app_id)
+
+      .then((app) => {
+        return app.connect_bucket(bucket_id);
+      })
 
-    api.apps3buckets.add(apps3bucket)
-      .then(() => {
-        res.redirect(routes.url_for('apps.details', { id: app_id }));
+      .then((_) => {
+        const { url_for } = require('../routes');
```

Comment:
See other comment on potentially moving this require on the top of the file if we use `url_for()` in different places.

Reply:
Ignore me.

Reply:
We can't do that, as explained